azurelinuxagent/common/logcollector.py

```diff
@@ -0,0 +1,377 @@
+            return path.replace(hardcoded_extension_log_path, _EXTENSION_LOG_DIR)
+        else:
+            return path
+
+    def _read_manifest_file(self):
+        with open(self._manifest_file_path, "rb") as in_file:
+            data = in_file.read()
+            if data is None:
+                return None
+            else:
+                data = ustr(data, encoding="utf-8")
+                return data.splitlines()
+
+    @staticmethod
+    def _process_ll_command(folder):
+        parametrized_folder = LogCollector._parametrize_filepath(folder)
+        LogCollector._run_shell_command(["ls", "-alF", parametrized_folder], log_output=True)
+
+    @staticmethod
+    def _process_echo_command(message):
+        _LOGGER.info(message)
+
+    @staticmethod
+    def _process_copy_command(path):
+        parametrized_path = LogCollector._parametrize_filepath(path)
+        file_paths = glob.glob(parametrized_path)
+        for file_path in file_paths:
+            _LOGGER.info(file_path)
+        return file_paths
+
+    @staticmethod
+    def _convert_file_name_to_archive_name(file_name):
+        # File name is the name of the file on disk, whereas archive name is the name of that same file in the archive.
+        # For non-truncated files: /var/log/waagent.log on disk becomes var/log/waagent.log in archive
+        # (leading separator is removed by the archive).
+        # For truncated files: /var/truncated/var/log/syslog.1 on disk becomes truncated_var_log_syslog.1 in archive.
+        if file_name.startswith(_TRUNCATED_FILES_DIR):
+            original_file_path = file_name[len(_TRUNCATED_FILES_DIR):].lstrip(os.path.sep)
+            archive_file_name = LogCollector._TRUNCATED_FILE_PREFIX + original_file_path.replace(os.path.sep, "_")
+            return archive_file_name
+        else:
+            return file_name.lstrip(os.path.sep)
+
+    @staticmethod
+    def _remove_uncollected_truncated_files(files_to_collect):
+        # After log collection is completed, see if there are any old truncated files which were not collected
+        # and remove them since they probably won't be collected in the future. This is possible when the
+        # original file got deleted, so there is no need to keep its truncated version anymore.
+        truncated_files = os.listdir(_TRUNCATED_FILES_DIR)
+
+        for file_path in truncated_files:
+            full_path = os.path.join(_TRUNCATED_FILES_DIR, file_path)
+            if full_path not in files_to_collect:
+                if os.path.isfile(full_path):
+                    os.remove(full_path)
+
+    @staticmethod
+    def _expand_parameters(manifest_data):
+        _LOGGER.info("Using {0} as $LIB_DIR".format(_AGENT_LIB_DIR))
+        _LOGGER.info("Using {0} as $LOG_DIR".format(_EXTENSION_LOG_DIR))
+        _LOGGER.info("Using {0} as $AGENT_LOG".format(_AGENT_LOG))
+
+        new_manifest = []
+        for line in manifest_data:
+            line.replace("$LIB_DIR", _AGENT_LIB_DIR)
```

Comment:
Don't you wanna replace the `$AGENT_LOG` too?

Reply:
Thanks, added!